### ISSUE TITLE
Getting innerHTML throw TypeError: Cannot call method 'toLowerCase' of undefined

### ISSUE THREAD

**Issue by RickyChien** [tmpvar/jsdom#1004]
Hi, I ran into a TypeError while parsing with ```<template>``` tag.

Here is my sample code
```
var jsdom = require('jsdom').jsdom;

var e = jsdom('<template><div></div></template>').documentElement.innerHTML;
```
above is fine.

```
var jsdom = require('jsdom').jsdom;

var e = jsdom('<template>    <div></div>    </template>').documentElement.innerHTML;
```

Second one will throw error. I found if there exists some white-spaces within ```<template></template>```, it throws:

```
TypeError: Cannot call method 'toLowerCase' of undefined
    at Object.exports.getTagName (/Users/Ricky/Documents/gaia/node_modules/jsdom/lib/jsdom/browser/documentAdapter.js:20:26)
    at Serializer._serializeTextNode (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:155:47)
    at Serializer._serializeChildNodes (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:76:22)
    at Serializer._serializeElement (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:116:14)
    at Serializer._serializeChildNodes (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:73:22)
    at Serializer._serializeElement (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:116:14)
    at Serializer._serializeChildNodes (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:73:22)
    at Serializer.serialize (/Users/Ricky/Documents/gaia/node_modules/jsdom/node_modules/parse5/lib/serialization/serializer.js:58:10)
    at exports.domToHtml (/Users/Ricky/Documents/gaia/node_modules/jsdom/lib/jsdom/browser/domtohtml.js:16:20)
```

**Comment by domenic** [comment]
What version of jsdom? There were some changes in this area between 2.0 and 3.0, so whichever one you are not testing would be good to test to figure out where the bug is :)

**Comment by RickyChien** [comment]
@domenic my version is jsdom@3.0.0

**Comment by domenic** [comment]
Fixed in 3.0.1 :)

**Comment by RickyChien** [comment]
@domenic wow you're so awesome!! It looks fine after updating to 3.0.1. Thank a lot
